python/lsst/pipe/supertask/cmdLineActivator.py

```diff
@@ -0,0 +1,554 @@
+#
+# LSST Data Management System
+# Copyright 2017 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+"""
+Module defining CmdLineActivator class and related methods.
+"""
+
+from __future__ import print_function
+from builtins import object
+
+#--------------------------------
+#  Imports of standard modules --
+#--------------------------------
+import sys
+import traceback
+
+#-----------------------------
+# Imports for other modules --
+#-----------------------------
+import lsst.log as lsstLog
+import lsst.pipe.base.argumentParser as pipeBaseArgParser
+from lsst.pipe.base.task import TaskError
+from .activator import profile
+from .parser import makeParser
+from .taskLoader import (TaskLoader, KIND_CMDLINETASK, KIND_SUPERTASK)
+
+
+#----------------------------------
+# Local non-exported definitions --
+#----------------------------------
+
+__all__ = ['CmdLineActivator']
+
+# logging properties
+_logProp = """\
+log4j.rootLogger=INFO, A1
+log4j.appender.A1=ConsoleAppender
+log4j.appender.A1.Target=System.err
+log4j.appender.A1.layout=PatternLayout
+log4j.appender.A1.layout.ConversionPattern={}
+"""
+
+
+def _table(tuples, headers):
+    """Nice formatting of 2-column table
+    """
+    if not tuples:
+        return
+    width = max(len(x[0]) for x in tuples)
+    if headers:
+        width = max(width, len(headers[0]))
+        print(headers[0].ljust(width), headers[1])
+        print(''.ljust(width, '-'), ''.ljust(len(headers[1]), '-'))
+    for col1, col2 in tuples:
+        print(col1.ljust(width), col2)
+
+#------------------------
+# Exported definitions --
+#------------------------
+
```

Comment:
I'll fix `_table()` documentation and naming.

Regarding private/public - having underscores for non-exported names is a good convention, independently of `__all__` being defined or not. In case `__all__` is defined it's a sort of reminder "don't add me to `__all__`". And `__all__` is not always defined, in fact I added `__all__` only after I finished writing this module. I'd prefer to not use `__all__` and depend entirely on underscore prefixes, but that does not always work as expected. The root of evil is `import *` of course and we should not be using it but it looks like it is a common pattern in `__init__.py` files.

Ordering of private/public in Python is a complicated issue. There are cases when public definitions depend on private things being already defined, and in that case the order has to be "private first". In my experience that order works better so I use that ordering (and LSST coding conventions are silent about that).

I'll move `__all__` about section comment to avoid confusion.